Alento Slash Commands

DenverCoder1 opened the pull request.

alentoghostflame committed:
Very ugly, very basic slash command setup for very ugly, very basic commands.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
register_application_commands() auto-happens on_connect rather than being manually called

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added super limited basic support for cogs

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Moved things to use a subclass of Client, basically dropped cog support for now.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Cleanup from different development directions.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed args typed with normal types not being optional by default

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed choices being broken, updated required logic

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added basic message and user command decorators

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Undid accidential deletion

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added support for channel_types

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
CommandArgument now verifies in init(), making errors more usable.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

veni-vidi-code committed:
Changes option discord from `View Documentation For` of the docs to nextcord (#182)

veni-vidi-code committed:
Fixed a typo in ui docs (#185)

* Fixed a typo

* Update item.py

DenverCoder1 commented on nextcord/application_command.py:
```diff
@@ -0,0 +1,97 @@
+    from .guild import Guild
+
+
+class ApplicationCommandType(IntEnum):
+    CHAT_INPUT = 1
+    USER = 2
+    MESSAGE = 3
+
+
+class ApplicationCommandOptionType(IntEnum):
+    SUB_COMMAND = 1
+    SUB_COMMAND_GROUP = 2
+    STRING = 3
+    INTEGER = 4
+    BOOLEAN = 5
+    USER = 6
+    CHANNEL = 7
+    ROLE = 8
+    MENTIONABLE = 9
+    NUMBER = 10  # A double, AKA floating point, AKA decimal.
```
These enums can go in [enums.py](https://github.com/nextcord/nextcord/blob/b644c29841ec878b77b6a38bdb6e754bf7ec90e2/nextcord/enums.py) like the others.

Try to keep the style consistent with existing stuff as well (eg. lowercase, inherit `Enum`)

DenverCoder1 commented on nextcord/application_command.py:
```diff
@@ -0,0 +1,97 @@
+        self.name: str = payload.get('name')
+        self.value: Union[str, int, float] = payload.get('value')
+
+    def from_data(self, data: dict):
+        self.__init__(data)
```
Is this used anywhere? Seems unecessary.

DenverCoder1 commented on nextcord/application_command.py:
```diff
@@ -0,0 +1,97 @@
+    from .guild import Guild
+
+
+class ApplicationCommandType(IntEnum):
+    CHAT_INPUT = 1
+    USER = 2
+    MESSAGE = 3
```
These are for Context Menus, right? Maybe name them something like `ContextMenuType`?

DenverCoder1 commented on nextcord/client.py:
```diff
@@ -1643,3 +1646,32 @@ def persistent_views(self) -> Sequence[View]:
+        await self.get_application_commands()
+
+    async def on_interaction(self, interaction: Interaction):
+        # TODO: This doesn't seem optimal, I need reassurance?
+        if interaction.type is InteractionType.ping:
+            self.dispatch("ping", interaction)
+        elif interaction.type is InteractionType.application_command:
+            self.dispatch("application_command", interaction)
+        elif interaction.type is InteractionType.component:
+            self.dispatch("component", interaction)
```
Possibly still could be done better, but some repetition could be avoided with a dict 

```suggestion
        events = {
        	InteractionType.ping: "ping",
        	InteractionType.application_command: "application_command",
        	InteractionType.component: "component"
        }
        self.dispatch(events[interaction.type], interaction)
```

DenverCoder1 commented on nextcord/command_client.py:
```diff
@@ -0,0 +1,396 @@
+            return ret
+
+
+class CommandClient(Client):
```
I'm not sure this is a good idea...

Making another Client class to do slash commands is the one thing everyone hated Pycord for.

ooliver1 replied:
I don’t know exactly but people don't like 3rd party libs because of that, however as a fork we have control over the code unlike a 3rdplib doesn't

shepherdjay committed:
Fixes #165 create_sticker (#189)

DenverCoder1 committed:
Added Halloween server icon (#198)

* docs(assets): Added Halloween server icon

* Delete nextcord-icon-halloween.svg

alentoghostflame committed:
Added basic application-command-specific CommandCog support.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added barely tested subclassing to Bot and Cogs

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

apat7 committed:
added spooktober discord banners (#192)

* added spooktober discord banners

* svg not showing up

* aligned up text

* changed font

* changed file name

SayHelloRoman committed:
replacing discord with nextcord (#205)

alentoghostflame committed:
Fixed some regressions, added some migratory helpers

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Some cleanup, moved enum to enums, user and message commands now receive Member and Message argument respectively.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

DenverCoder1 committed:
Renamed assets for consistency (#203)

* docs(assets): renamed assets

* docs(assets): added halloween repo banner

* docs(assets): fix naming again

* docs(readme): fix naming in readme

* docs(assets): Removed halloween repo banner

https://github.com/nextcord/nextcord/pull/202#issuecomment-936031646

rf20008 committed:
Small grammar fix in docstring (#210)

alentoghostflame committed:
Changed inits for lists and dicts as requested, renamed the badly named CmdArg to SlashOption

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

ooliver1 committed:
Test 1 to fix #218 (#219)

alentoghostflame committed:
Reworked how commands are registered. Client now holds ApplicationCommands and bulk registers somewhat efficiently.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
typing as Member now returns User if there isn't a Guild involved.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added imports to __init__.py

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added potential fix for getting user data in DM's

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Typing as message now converts to int

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added some documentation, is this useful?

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added support for a slash, user, and message command having duplicate names in bulk creation

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Removed no longer used args for CommandClient

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed subcommands not being registered properly.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

drlove2002 committed:
Added type hint for avatar_url (#221)

alentoghostflame committed:
Purposefully broke registartion of commands. Added lazy loading of application commands.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

VincentRPS committed:
Update client.py

VincentRPS committed:
Update client.py

alentoghostflame committed:
Merge pull request #4 from VincentRPS/patch-4

Made Better Errors

alentoghostflame committed:
Renamed the argument handler and added stubs for future work

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added better checks to see if interaction can work with an already registered command.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
More work on rollout, added registration of global commands. Needs bug fixing.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added basic guild rollout, added deeper payload check for guild and global commands.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed subcommand groups

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

VincentRPS committed:
Made README More Modern

VincentRPS committed:
Fix

VincentRPS committed:
Fix X2

VincentRPS committed:
Fixed Link

VincentRPS committed:
Update README.rst

Co-authored-by: Jonah Lawrence <jonah@freshidea.com>

VincentRPS committed:
Update README.rst

Co-authored-by: Jonah Lawrence <jonah@freshidea.com>

VincentRPS committed:
Removed Buttons

VincentRPS committed:
Update README.rst

Co-authored-by: Jonah Lawrence <jonah@freshidea.com>

VincentRPS committed:
Update README.rst

VincentRPS committed:
Update README.rst

VincentRPS committed:
Update README.rst

VincentRPS committed:
Update README.rst

squigjess committed:
on_voice_state_update now cleans up broken vc connections when forcibly disconnected by user (#222)

* on_voice_state_update now cleans up broken vc connections

* reverting previous commit

* forced disconnects now call cleanup() as intended

VincentRPS committed:
Update README.rst

alentoghostflame committed:
Merge pull request #5 from VincentRPS/patch-6

Made README More Modern

alentoghostflame committed:
Turned option_types into a dictionary, added some comments.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed Message and User commands needlessly re-registering, spruced up print statements.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

Commit:
fix: missing required `py.typed` for type checkers (#236)

joymalya31 committed:
updated (#237)

alentoghostflame committed:
Some cleanup, added options to Client.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Restored README to its old state for now

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Some cleanup, added check when making subcommands.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
First round of documentation

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Second round of documention, further cleanup.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

VincentRPS committed:
Make Discord Server Info Better An Higher Quality With Shields.io (#239)

* Make Discord Server Info Better An Higher Quality With Shields.io

* Changed Stuff

* Update README.rst

* Changed Back To Blue

DenverCoder1 committed:
Adds November themed server assets (#240)

* Squashed commit of the following:

commit 703e3d75b24c3ac7c28d530cdc981236d982db04
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Mon Nov 1 12:48:24 2021 +0200

    Adds November themed server assets

commit beb4368761804939ad52e43cf63d90fb2e410d36
Merge: 88241aec 99d2312f
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Mon Nov 1 12:11:09 2021 +0200

    Merge branch 'nextcord:master' into master

commit 88241aec92db1208db09bb10f11d84f467622c34
Merge: d228235e 21fd80ca
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 13 12:41:30 2021 +0300

    Merge branch 'nextcord:master' into master

commit d228235e36b797d1fd44b1f805e568cd740ad388
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 14:26:25 2021 +0300

    docs(assets): Removed halloween repo banner

    https://github.com/nextcord/nextcord/pull/202#issuecomment-936031646

commit 08f6de25fa78a226811cd2ca68dd273fbeefc4f3
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 13:41:28 2021 +0300

    docs(readme): fix naming in readme

commit 65f0c16c41340d059f38547a4f6ade0d1060f43f
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 10:39:42 2021 +0000

    docs(assets): fix naming again

commit 07ee9b9d29d9fdab22468c37682fdc0db620e3f0
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 13:31:22 2021 +0300

    docs(assets): added halloween repo banner

commit b495b1ba96f125eccc24a3d60680ab6d1efa6a56
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 10:30:12 2021 +0000

    docs(assets): renamed assets

commit 87b0aee1b9ec093f271b61cf32f3f8ba932a5b23
Merge: c1c7c30a ef5d01d6
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 13:27:24 2021 +0300

    Merge branch 'nextcord:master' into master

commit c1c7c30a154209f70c5e98474a370374a3004849
Merge: da314d84 75c9d1f7
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Tue Oct 5 00:07:32 2021 +0300

    Merge branch 'nextcord:master' into master

commit da314d84641b898694348162e5dd63940e0906c2
Merge: 84b319bd 5fd0a213
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Fri Oct 1 01:56:08 2021 +0300

    Merge pull request #1 from alentoghostflame/master

* Adds November themed server assets

DenverCoder1 committed:
Updates November server icon (#241)

* Squashed commit of the following:

commit 703e3d75b24c3ac7c28d530cdc981236d982db04
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Mon Nov 1 12:48:24 2021 +0200

    Adds November themed server assets

commit beb4368761804939ad52e43cf63d90fb2e410d36
Merge: 88241aec 99d2312f
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Mon Nov 1 12:11:09 2021 +0200

    Merge branch 'nextcord:master' into master

commit 88241aec92db1208db09bb10f11d84f467622c34
Merge: d228235e 21fd80ca
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 13 12:41:30 2021 +0300

    Merge branch 'nextcord:master' into master

commit d228235e36b797d1fd44b1f805e568cd740ad388
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 14:26:25 2021 +0300

    docs(assets): Removed halloween repo banner

    https://github.com/nextcord/nextcord/pull/202#issuecomment-936031646

commit 08f6de25fa78a226811cd2ca68dd273fbeefc4f3
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 13:41:28 2021 +0300

    docs(readme): fix naming in readme

commit 65f0c16c41340d059f38547a4f6ade0d1060f43f
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 10:39:42 2021 +0000

    docs(assets): fix naming again

commit 07ee9b9d29d9fdab22468c37682fdc0db620e3f0
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 13:31:22 2021 +0300

    docs(assets): added halloween repo banner

commit b495b1ba96f125eccc24a3d60680ab6d1efa6a56
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 10:30:12 2021 +0000

    docs(assets): renamed assets

commit 87b0aee1b9ec093f271b61cf32f3f8ba932a5b23
Merge: c1c7c30a ef5d01d6
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Wed Oct 6 13:27:24 2021 +0300

    Merge branch 'nextcord:master' into master

commit c1c7c30a154209f70c5e98474a370374a3004849
Merge: da314d84 75c9d1f7
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Tue Oct 5 00:07:32 2021 +0300

    Merge branch 'nextcord:master' into master

commit da314d84641b898694348162e5dd63940e0906c2
Merge: 84b319bd 5fd0a213
Author: Jonah Lawrence <jonah@freshidea.com>
Date:   Fri Oct 1 01:56:08 2021 +0300

    Merge pull request #1 from alentoghostflame/master

* Adds November themed server assets

* Updates November server icon

alentoghostflame committed:
Renamed CommandCog to ClientCog, migrated application command decorators to Client, removed CommandClient, moved everything remaining from nextcord.command_client to nextcord.application_command

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Removed unneeded file

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Alphabetical Imports.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Renamed some variables to reference the current name of ClientCog, added some in-code documentation.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Moved processing of Application Commands to its own function, added defaults in docstring of perform_application_command_rollout

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
User commands now add user payload to cache, User and Message commands now use resolved payloads when needed, added comments.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

Commit:
Bump Python Version (#254)

Commit:
Changed Capitalization to be better (#245)

Labels in README are all lowercase for consistency

alentoghostflame committed:
Moved global rollout to on_connect and guild rollouts to on_guild_available

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added single-command registration

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed typo in logging, fixed SlashOption with requires set to False triggering re-registration of the command.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed coro tags as per oolivers request, updated note in commands.ext.bot as per Skelmis's request.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed import order, added typings to functions/methods.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Oh right licensing exists.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

VincentRPS committed:
Adding 3.10 in supported python versions (#255)

* Adding 3.10 in supported python versions

* Adding 3.10 As Supported In unaliased

veni-vidi-code committed:
Create CODEOWNERS (#242)

* Create CODEOWNERS

First Codeowners

* Update CODEOWNERS

* Update CODEOWNERS

* Update CODEOWNERS

* Update CODEOWNERS

Clearer instructions on how to apply

* Fix CODEOWNERS

Co-authored-by: RPS <vincentbusiness55@gmail.com>

ThePrivatePanda committed:
Fix embed.url description. (#273)

* Change Embed.url description. 

Embed.url was defined as 'The url of the embed', which incorrectly described its usage as to hyperlink the title of the embed.

* Update embeds.py

Fix grammar or language or whatever you would like to call it to avoid ugly repetition.

Skelmis committed:
Feature: Cog's can now have external config (#233)

* Bump release level

* Feature: #129

* Implement runtime type checking

* Derive typechecking from ExtensionFailed

* Resolve dzshn reviews

* Remove version bump

* Resolve ooliver's review

* Add new exception to __all__

* Update nextcord/ext/commands/bot.py

Co-authored-by: Sofia "dzshn" N. L. <zshn@pm.me>

* Resolve issue

Co-authored-by: Sofia "dzshn" N. L. <zshn@pm.me>

Skelmis committed:
Feature: Group's hook & check inheritance (#238)

* Added a way for group commands to inherit handlers

#123

* Fix issues with parents not having invokes

* Add default

* Some minor changes to return values

* Fix review

* Slight wording change

alentoghostflame committed:
Large redoing of some base portions of ApplicationCommands. Syntax of decorator usage should be the same, but they should be slightly easier to create during runtime.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

DenverCoder1 committed:
Adds December themed server icon and banner (#275)

* chore(assets): Adds December icon and banner

alentoghostflame committed:
Increased logging slightly

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed kwargs having no effect on application (sub[-sub])commands, added untested SlashOption params, added verification checks, added more MISSINGs, added beginning of on_autocomplete verification.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed SlashOption regression, unfinished autocomplete work.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added basic autocomplete support.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Added docstring to send_autocomplete

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Now autoenables autocomplete if you set an autocomplete function without explicitly disabling autocomplete.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Apparently Discord doesn't send the focused keyword if it isn't actually focused. Fixed KeyError related to that.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Commented out ApplicationCommandResponse, reorganized methods inside of ApplicationSubcommand and ApplicationCommand, added some comments.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

VincentRPS committed:
Adding Framework Classifiers (#283)

alentoghostflame committed:
Logging changes, removed more restrictions related to the bot stopping you from registering commands that don't have explicit Guild ID's set

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

parnexcodes committed:
fix typo (#285)

Nymfya committed:
Fix: MessageConverter error when UserInput only provide MessageID (#276)

VincentRPS committed:
Formatting per-commit ci/cd (#281)

add formatting gh action

ooliver1 committed:
Add interaction.send() (#234)

* add interaction.send: a fallback method; with docs

* sphinx docs seem to be incorrect, there is no :property:, anyway fixed typos

* refactor(interaction): forgot it needs to return

to receive message object

* feat(inter): interaction.edit, like send

Suven-p committed:
Add webhook example (#211)

* Add webhook example

* Remove bot from webhook example

* Remove sync example

* Remove unused requests

* Run function

* Remove variables

* Add comments

* Update examples/webhook.py

its quite stale so I'll just add it

Co-authored-by: Oliver Wilkes <34910574+ooliver1@users.noreply.github.com>

veni-vidi-code committed:
Bugfix for task.loop (#201)

Makes task.loop check if the time must be on this day or the next

TAG-Epic committed:
Fix: Bump aiohttp version (#259)

vcokltfre committed:
fix: use correct types for channel events (#287)

fix: use correct types for channel events

veni-vidi-code committed:
Update core.py (#288)

Throw a warning when docbuilding otherwise

alentoghostflame committed:
Moved ApplicationCommand objects into ConnectionState, changed how guild ID's to rollout are added.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Small patch towards command registration during runtime not getting deleted by the deletion routine.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

ooliver1 committed:
feat: bump to a4 (#291)

alentoghostflame committed:
Fixed autocomplete, did I start working on it then forget what I was doing?

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Changed the ordering that application commands are rolled out in, changed function locations.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Fixed deep checks not actually being all that deep.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
General cleanup, added a few docstrings and methods

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Small removal of commented code

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Merge remote-tracking branch 'origin/master' into alento_slash_commands

alentoghostflame committed:
Added basic docstrings for application command decorators and ApplicationCommand.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

ThePrivatePanda committed:
Update styles.css (#294)

ooliver1 committed:
Timeout permission, modify and property (#292)

Co-authored-by: Epic <tagepicuwu@gmail.com>

alentoghostflame committed:
Didn't take into account that ConnectionState clears itself after commands decorated with Client/Bot are added. Fixed.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Moved an exception to nextcord.errors, addressed most of Skelmis's review.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Made several functions look visually better.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

alentoghostflame committed:
Changed InvalidCommandType to subclass a more relevent exception.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

VincentRPS committed:
Add `InvalidCommandType` in the exception hierarchy

alentoghostflame committed:
Merge pull request #9 from VincentRPS/patch-14

Add `InvalidCommandType` in the exception hierarchy

alentoghostflame committed:
Removed Optional from SlashOption default typing.

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

leestarb committed:
Fix double "@" in Guild.default_role.mention (#300)

ooliver1 committed:
`Permissions.all()` uses `-1` (#305)

This does work, yes, but this should fix an issue with embedded activities and moderate members permissions

vcokltfre committed:
Application Command Support

feat: add application command support

alentoghostflame committed:
Merge remote-tracking branch 'origin/master' into alento_slash_commands

alentoghostflame committed:
Added basic slash examples

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>
